src/atomic_file/atomic.rs

```diff
@@ -92,37 +94,32 @@ pub struct AtomicFile {
     prefix: String,
 }
 
-fn parse_version(filename: &std::ffi::OsStr, prefix: &str) -> Option<usize> {
+fn parse_version(filename: &std::ffi::OsStr) -> Option<usize> {
     let filename = filename.to_str()?;
-    if !filename.starts_with(prefix) {
-        return None;
-    }
-    filename[prefix.len()..].parse().ok()
+    let (_, version) = filename.rsplit_once('.')?;
+    version.parse().ok()
 }
 
 impl AtomicFile {
-    pub fn new(path: impl Into<PathBuf>) -> Result<Self> {
+    pub fn new(path: impl Into<PathBuf>) -> crate::Result<Self> {
         let directory = path.into();
+        let machine_id = machine_uid::get()?;
         std::fs::create_dir_all(&directory)?;
         let filename: &str = match directory.file_name() {
             Some(name) => name.to_str().unwrap(),
-            None => {
-                return Err(std::io::Error::new(
-                    std::io::ErrorKind::InvalidInput,
-                    "`path` must specify a directory name",
-                ));
-            }
+            None => Err(std::io::Error::new(
+                std::io::ErrorKind::InvalidInput,
+                "`path` must specify a directory name",
+            ))?,
         };
-        let prefix = format!("{}.", filename);
+        let prefix = format!("{}_{}.", filename, machine_id);
```

Comment:
> In Linux, machine id is a single newline-terminated, hexadecimal, 32-character, lowercase ID. When decoded from hexadecimal, this corresponds to a 16-byte/128-bit value. This ID may not be all zeros. This ID uniquely identifies the host. It should be considered “confidential”, and must not be exposed in untrusted environments. And do note that the machine id can be re-generated by root.

We should hash the id for privacy reasons.

Reply:
Alternative is generating random unique id and saving it into local app data folder.

Reply:
The man pages say to hash it for direct use, so SHA256(machine_id) should be fine.

Otherwise yeah UUID4 works well, there's also UUID7 now, which is also random enough plus time sortable.

Reply:
https://www.freedesktop.org/software/systemd/man/machine-id.html

Reply:
Ok for not using the machine id. However where do you want to store the file containing the Id? Since it's a lib we can pass a variable with the root parameter but don't know which constructor do you want for that. however not sure if we want to pass to the machine identifier on the constructor of the AtomicFile or somewhere else

Reply:
- #54
- #55